### PR TITLE
SONARJAVA-2177: @RequestMapping methods should be public

### PR DESCRIPTION
**Pull request by agigleux**

### CODE REVIEW

File: java-checks/src/test/files/checks/spring/RequestMappingMethodPublicCheck.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class NonSpringComponentClazz {
+  @RequestMapping(value = "/greet", method = GET)
+  private String greet(String greetee) { // Compliant
+  }
+
+  @RequestMapping(value = "/hello", method = RequestMethod.GET)
+  public String hello(String greetee) { // Compliant
+  }
+}
```
Comment (m-g-sonar) [inline java-checks/src/test/files/checks/spring/RequestMappingMethodPublicCheck.java]: missing new line.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S3751_java.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+"status": "ready",
+"remediation": {
+"func": "Constant/Issue",
+"constantCost": "2min"
+},
+"tags": [
+"spring"
+],
+"defaultSeverity": "Blocker"
+}
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S3751_java.json]: missing new line.

File: java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  private static boolean isClassController(Symbol.MethodSymbol methodSymbol) {
+    SymbolMetadata parentClassOwner = methodSymbol.owner().metadata();
+    return parentClassOwner.isAnnotatedWith("org.springframework.stereotype.Controller");
+  }
+
+  private static boolean isRequestMappingAnnotated(MethodSymbol methodSymbol) {
+    return methodSymbol.metadata().isAnnotatedWith("org.springframework.web.bind.annotation.RequestMapping");
+  }
+
+  private static boolean isPublicMethod(MethodSymbol methodSymbol) {
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java]: not sure you need a dedicated method for this one, as it's pretty obvious.

File: java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    if (isClassController(methodSymbol)
+      && isRequestMappingAnnotated(methodSymbol)
+      && !isPublicMethod(methodSymbol)) {
+      reportIssue(methodTree.simpleName(), "Make this method \"public\".");
+    }
+  }
+
+  private static boolean isClassController(Symbol.MethodSymbol methodSymbol) {
+    SymbolMetadata parentClassOwner = methodSymbol.owner().metadata();
+    return parentClassOwner.isAnnotatedWith("org.springframework.stereotype.Controller");
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java]: you can inline the parentClassOwner(metadata)

File: java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ */
+package org.sonar.java.checks.spring;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Symbol.MethodSymbol;
+import org.sonar.plugins.java.api.semantic.SymbolMetadata;
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java]: MINOR Remove this unused import 'org.sonar.plugins.java.api.semantic.SymbolMetadata'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ */
+package org.sonar.java.checks.spring;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Tree;
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/spring/RequestMappingMethodPublicCheck.java]: MINOR Remove this unused import 'org.sonar.plugins.java.api.semantic.SymbolMetadata'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Pull request merged**